### PR TITLE
Explicitly specify version of System.Net.Http

### PR DESCRIPTION
**Pull request by Jamminroot**
Things to consider before you submit the PR:

* [x] Has `CHANGELOG.md` been updated?
* [ ] Are tests passing locally?
* [x] Are the files formatted correctly?
* [ ] Did you add unit tests if this modifies the Windows code?
* [ ] Did you test your change with either the sample apps that are included in the repository or with a blank app that uses your change?

## Description

This PR makes sure that System.Net.Http library is of correct version.

## Related PRs or issues

[AB#86366](https://msmobilecenter.visualstudio.com/454a20a9-dfe6-4c1f-8ae8-417f76adb472/_workitems/edit/86366)

## Misc

Add what's missing, notes on what you tested, additional thoughts or questions.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,6 +2,10 @@
 
 ## Version 4.2.1 (Under development)
 
+### AppCenter
+
+* **[Fix]** System.Net.Http version is locked at `4.3.4` in order to mitigate .NET Core Information Disclosure Vulnerability.
```
Comment (MatkovIvan) [inline CHANGELOG.md:7]: This affects only tests/demos.  No update in SDK and changeelog

**Review by MatkovIvan — CHANGES_REQUESTED** [review]

File: SDK/AppCenter/Microsoft.AppCenter.WindowsDesktop/Microsoft.AppCenter.WindowsDesktop.csproj [hunk 2]
```diff
@@ -18,6 +18,7 @@
   <ItemGroup>
     <PackageReference Include="Newtonsoft.Json" Version="12.0.2" />
     <PackageReference Include="SQLitePCLRaw.bundle_green" Version="2.0.2" />
+    <PackageReference Include="System.Net.Http" Version="4.3.4" />
```
Comment (MatkovIvan) [inline SDK/AppCenter/Microsoft.AppCenter.WindowsDesktop/Microsoft.AppCenter.WindowsDesktop.csproj:21]: There was no warning for this sub-project (SDK part). Why did you add extra dependency here?

Comment (Jamminroot) [reply SDK/AppCenter/Microsoft.AppCenter.WindowsDesktop/Microsoft.AppCenter.WindowsDesktop.csproj:21]: Those are the packages \ sub-projects using System.Net.Http; I think fixing all the related places would be a good call, rather than those which were detected.

**Comment by Jamminroot** [comment]
Closed PR as detection is false-positive. Confirmed on call with @MatkovIvan
